suite-native/receive/src/components/DeviceScreenContent.tsx

```diff
@@ -0,0 +1,165 @@
+import { useSelector } from 'react-redux';
+
+import {
+    Canvas,
+    Group,
+    Text as SkiaText,
+    LinearGradient,
+    useFont,
+    vec,
+} from '@shopify/react-native-skia';
+
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+import { DeviceModelInternal } from '@trezor/connect';
+import { selectDeviceModel } from '@suite-common/wallet-core';
+
+import { DeviceScreenPagination } from './DeviceScreenPagination';
+import { DEVICE_SCREEN_BACKGROUND_COLOR, DEVICE_TEXT_COLOR } from '../constants';
+import { parseAddressToDeviceLines } from '../utils';
+import { DevicePaginationActivePage, isPaginationCompatibleDeviceModel } from '../types';
+
+type DeviceScreenContentProps = {
+    address: string;
+    isAddressRevealed: boolean;
+    isPaginationEnabled: boolean;
+    activePage: DevicePaginationActivePage;
+};
+
+const T2T1_PAGER_OFFSET = 60;
+const T2B1_PAGER_OFFSET = 40;
+
+const deviceToContentStyles = {
+    [DeviceModelInternal.T1B1]: {
+        fontSource: require('../../../../packages/theme/fonts/PixelOperatorMono8-Regular.ttf'),
+        fontSize: 14,
+        lineWidth: 295,
+        lineHeight: 17,
+    },
+    [DeviceModelInternal.T2T1]: {
+        fontSource: require('../../../../packages/theme/fonts/RobotoMono-Regular.ttf'),
+        fontSize: 20,
+        lineWidth: 215,
+        lineHeight: 25,
+    },
+    [DeviceModelInternal.T2B1]: {
+        fontSource: require('../../../../packages/theme/fonts/PixelOperatorMono8-Regular.ttf'),
+        fontSize: 14,
+        lineWidth: 265,
+        lineHeight: 25,
+    },
+} as const satisfies Record<
+    DeviceModelInternal,
+    { fontSource: string; fontSize: number; lineWidth: number; lineHeight: number }
+>;
+
+const contentCanvasStyle = prepareNativeStyle<{
+    lineWidth: number;
+    lineHeight: number;
+    numberOfLines: number;
+    isPaginationEnabled: boolean;
+    deviceModel: DeviceModelInternal;
+}>((_, { lineWidth, lineHeight, isPaginationEnabled, deviceModel, numberOfLines }) => ({
+    width: lineWidth,
+    height: lineHeight * numberOfLines,
+
+    extend: [
+        {
+            // IF pagination is enabled, the content canvas should be 4 lines high
+            // so the content of the screen does not jump while switching pages.
+            condition: isPaginationEnabled,
+            style: {
+                height: lineHeight * 4,
+            },
+        },
+        {
+            condition: isPaginationEnabled && deviceModel === 'T2T1',
+            style: {
+                width: lineWidth + T2T1_PAGER_OFFSET,
```

Comment:
Create some object like `deviceToContentStyles` and map each device to it's offset there.

Reply:
fixed 45dd281869